superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx

```diff
@@ -244,9 +244,23 @@ class DatasourceControl extends React.PureComponent {
     return (
       <Styles data-test="datasource-control" className="DatasourceControl">
         <div className="data-container">
-          <Icons.DatasetPhysical className="dataset-svg" />
-          {/* Add a tooltip only for long dataset names */}
-          {!isMissingDatasource && datasource.name.length > 25 ? (
+          {isValidDatasourceType ? (
+            <Icons.ConsoleSqlOutlined className="datasource-svg" />
+          ) : (
+            <Icons.DatasetPhysical className="datasource-svg" />
+          )}
+          {isValidDatasourceType ? (
```

Comment:
we need to check if the datasource.type is a `DatasourceType.Query` here, if so then show the image in sql as the name

Reply:
I changed this to be a strict check for just Query in [`this commit`](https://github.com/apache/superset/pull/20317/commits/6fab57f9ddac90ed6a4aefe028ea1f66905858d0) 😁